src/Thumbhash.php

```diff
@@ -2,8 +2,101 @@
 
 namespace Thumbhash;
 
-class Thumbhash
+final class Thumbhash
 {
+
+    private $content;
+    private $width;
+    private $height;
+    private $pixels;
+    private $hash;
+    private $key;
+    private $url;
+
+    function __construct(string $file)
+    {
+        
+        $this->content = file_get_contents($file);
+        if (class_exists('Imagick')) {
+            list($this->width, $this->height, $this->pixels) = $this->extract_size_and_pixels_with_imagick($this->content);
+        } else {
+            list($this->width, $this->height, $this->pixels) = $this->extract_size_and_pixels_with_gd($this->content);
+        }
+
+        $this->hash = $this->RGBAToHash($this->width, $this->height, $this->pixels);
+        $this->key = $this->convertHashToString($this->hash); // You can store this in your database as a string
+    }
```

Comment:
I'm not against a real object but this constructor is wrong for a few reasons

1. Lot of framework uses [Flysystem](https://flysystem.thephpleague.com/docs/ ) abstraction to interact with files across different filesystems. [Laravel](https://laravel.com/docs/10.x/filesystem) even add a layer on top of that. I can also go further, what if my file comes from a stream ? This library shouldn't be responsible of file processing.
2. Actually this class is "unidirectional", we can't take base64 hash and decode the image.